c10/xpu/XPUGuard.h

```diff
@@ -0,0 +1,255 @@
+#pragma once
+
+#include <c10/core/DeviceType.h>
+#include <c10/core/impl/InlineDeviceGuard.h>
+#include <c10/core/impl/InlineStreamGuard.h>
+#include <c10/xpu/XPUMacros.h>
+#include <c10/xpu/impl/XPUGuardImpl.h>
+
+namespace c10::xpu {
+
+/// A variant of DeviceGuard that is specialized for XPU.
+struct XPUGuard {
```

Comment:
We shouldn't introduce this and use `c10::DeviceGuard` directly

Reply:
refer to https://github.com/pytorch/pytorch/blob/9f8c84a39917a0aeafa415ede0e5e4efdae1b494/c10/cuda/CUDAGuard.h#L16-L21
`XPUGuard` has a slight difference with `DeviceGuard`. It is more efficient. And we intend to have our APIs align one-to-one with CUDA, saving the effort for users to migrate their code.

Reply:
As I was mentioning during the meeting. We do not want to migrate code by adding SomeDeviceFoo code next to the current CUDAFoo code. The migration will have to migrate the current CUDAFoo code to now use the generic Foo.
Unless we have end to end examples where this makes a meaningful perf difference, I don't think the extra maintenance burden is worth it.

Reply:
I got your point. Thanks for your elaboration.

Reply:
Remove the `XPUGuard`, `OptionalXPUGuard`, `XPUStreamGuard` and `OptionalXPUStreamGuard`.